src/utils/sortFeaturedPacks.js

```diff
@@ -0,0 +1,9 @@
+export const sortFeaturedPacks = (packs) => {
+  const featuredPacks = packs.filter((pack) =>
+    pack.keywords?.includes('featured')
+  );
+  const nonFeaturedPacks = packs.filter(
+    (pack) => !pack.keywords?.includes('featured')
+  );
+  return [...featuredPacks, ...nonFeaturedPacks];
+};
```

Comment:
I think we can probably use [`Array.prototype.sort()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) to handle this, rather than doing two passes through the list. `O(n)` is more performant than `O(2n)`. Here's how that could look:

```js
const isFeatured = (quickstart) =>
  quickstart.keywords && quickstart.keywords.includes('featured');

const sortedQuickstarts = quickstarts.sort((a, b) => {
  if (isFeatured(a) && !isFeatured(b)) {
    return 1;
  }
  if (isFeatured(b) && !isFeatured(a)) {
    return -1;
  }
  return 0;
});
```

In general, I like to favor built-in methods that allow the code to be a little more declarative about the operation we're trying to perform. We're trying to sort a list, so we use the `sort` method.

Also, I'm not sure if we have access to optional chaining (e.g. `foo?.bar?.baz`).

Also Also, we should really avoid references to "packs". This will help when we go to production readiness and need to update every instance to be "quickstarts".

Reply:
ahhh yes that makes more sense, i thought I couldn't use a sort but I did not think about using `isFeatured` as a boolean :aha: